### PR TITLE
Update grid to re-render when a checkbox item is checked

### PR DESCRIPTION
**Pull request by mmmmillar**
**What**  
Update grid to re-render when a checkbox item is checked

**Why**  
We have some features coming which need to update the page when a checkbox is checked

### CODE REVIEW

File: components/Feature/VulnerabilitiesGrid/index.js [hunk 1]
```diff
@@ -37,26 +54,32 @@ const VulnerabilitiesGrid = ({ onUpdate }) => {
...
+                    <Checkbox
+                      key={`${id}-v-${j}`}
+                      label={text}
+                      name={`${id}-v-${j}`}
+                      onClick={() =>
+                        updateVulnerabilities(`${id}-v-${j}`, text)
+                      }
+                    />
+                  );
+                })}
```
Comment (t-davies) [inline components/Feature/VulnerabilitiesGrid/index.js:68]: Not really a comment, just curious as to why you swapped from `(<... />)` to `{ return (<... />) }`? Was it just a readability thing?

File: components/Feature/VulnerabilitiesGrid/index.js [hunk 2]
```diff
@@ -10,16 +10,33 @@ import groups from './grid.json';
...
   const [grid, setGrid] = useState({ assets: {}, vulnerabilities: {} });
 
+  const addItem = (obj, key, value) => {
+    obj[key] = value;
+    return obj;
+  };
+
+  const removeItem = (obj, key) => {
+    return Object.fromEntries(Object.entries(obj).filter(([k]) => k != key));
+  };
```
Comment (t-davies) [inline components/Feature/VulnerabilitiesGrid/index.js:20]: If you modelled this as a `Map` you could do this slightly quicker, doing `fromEntries` on an `entries` that is filtered will iterate through all the keys maybe up to 3 times.

Probably doesn't really matter given the size of the objects we will be working with, but throwing it out there if you're interested.

Here's a slightly simplified example...

```
  // assuming we can turn assets into a map when we initialise the grid state
  const assets = new Map(Object.entries(someObjectOfInitialAssets));

  // add an item
  assets.set(key, value);

  // remove an item
  assets.delete(key);
```

**Review by t-davies — APPROVED** [review]

File: components/Feature/VulnerabilitiesGrid/index.js (hunk 1 above)
Comment (mmmmillar) [reply components/Feature/VulnerabilitiesGrid/index.js:68]: Just getting it ready for a change Bogdan is going to make

File: components/Feature/VulnerabilitiesGrid/index.js (hunk 2 above)
Comment (mmmmillar) [reply components/Feature/VulnerabilitiesGrid/index.js:20]: Nice, I might refactor this tonight

**Pull request merged**